### ISSUE TITLE
Impossible happened when upgrating to 1.3

### ISSUE THREAD

**Issue by AlexeyRaga** [commercialhaskell/stack#2841]
### General summary/comments (optional)
Upgrading stack from `1.2.0` to `1.3.0` fails on Mac OS 'Sierra'

### Steps to reproduce
- Run `stack upgrade`

### Expected
Stack upgrades itself to the latest version

### Actual
```
....
Preprocessing library stack-1.3.0...
[  1 of 121] Compiling Text.PrettyPrint.Leijen.Extended ( src/Text/PrettyPrint/Leijen/Extended.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/Text/PrettyPrint/Leijen/Extended.o )
[  2 of 121] Compiling Stack.Ghci.Script ( src/Stack/Ghci/Script.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/Stack/Ghci/Script.o )
[  3 of 121] Compiling Stack.FileWatch  ( src/Stack/FileWatch.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/Stack/FileWatch.o )
[  4 of 121] Compiling System.Process.PagerEditor ( src/System/Process/PagerEditor.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/System/Process/PagerEditor.o )
[  5 of 121] Compiling System.Process.Log ( src/System/Process/Log.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/System/Process/Log.o )
[  6 of 121] Compiling Paths_stack      ( .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/autogen/Paths_stack.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/Paths_stack.o )
[  7 of 121] Compiling Path.Find        ( src/Path/Find.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/Path/Find.o )
[  8 of 121] Compiling Path.Extra       ( src/Path/Extra.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/Path/Extra.o )
[  9 of 121] Compiling System.Process.Read ( src/System/Process/Read.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/System/Process/Read.o )
ghc: panic! (the 'impossible' happened)
  (GHC version 7.10.3 for x86_64-apple-darwin):
        Loading temp shared object failed: dlopen(/var/folders/mc/jmny0bj93c3998gzbf9m3qd40000gn/T/ghc58464_0/libghc_57.dylib, 5): no suitable image found.  Did find:
        /var/folders/mc/jmny0bj93c3998gzbf9m3qd40000gn/T/ghc58464_0/libghc_57.dylib: malformed mach-o: load commands size (47888) > 32768

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug

Completed 22 action(s).

--  While building package stack-1.3.0 using:
      /private/var/folders/mc/jmny0bj93c3998gzbf9m3qd40000gn/T/stack-upgrade56407/stack-1.3.0/.stack-work/dist/x86_64-osx/Cabal-1.22.5.0/setup/setup --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.5.0 build lib:stack exe:stack --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
```

### Stack version
```
$ stack --version
Version 1.2.0 x86_64 hpack-0.14.0
```

### Method of installation
* Official binary, downloaded from stackage.org or fpcomplete's package repository

**Comment by borsboom** [comment]
Duplicate of #2577

**Comment by borsboom** [comment]
TL;DR: macOS Sierra broke GHC.  GHC 8.0.2 will include a fix.  For now, upgrade by downloading a new binary.
